embc-app/ClientApp/src/app/provincial-admin/components/volunteer-organization-list/volunteer-organization-list.component.ts

```diff
@@ -82,144 +71,79 @@ export class VolunteerOrganizationListComponent implements OnInit, OnDestroy {
+    this.previousQuery = this.copyProperties(this.defaultSearchQuery);
+    // collect all volunteers
+    this.getVolunteers();
+    if (this.uniqueKeyService.getKey()) {
+      this.organizationService.getOrganizationById(this.uniqueKeyService.getKey()).subscribe(o => {
+        this.currentOrganization = o;
       });
+    } else {
+      // TODO: when the user gets kicked out of the organization for making an edit redirect them back to the place they can make a decision
+      this.router.navigate([`/${this.path}/organizations`]);
+    }
   }
 
   ngOnDestroy() {
     // close modal if it's open
     if (this.confirmModal) { this.confirmModal.dismiss(); }
   }
-
-  initSearchForm(): void {
-    // instantiate the form builder group
-    this.form = this.fb.group({
-      // clear the search box
-      searchbox: null,
-      // set the user toggle to a constant integer that defines the view for the user
-      userToggle: this.SHOW_ALL,
-    });
-  }
-
-  initVars(): Observable<Organization> {
-    // combine everything we need to display this form into one Observable
-    const combined$ = combineLatest(
-      this.auth.getCurrentUser(),
-      this.auth.isLocalAuthority$,
-      this.auth.isProvincialAdmin$,
-      this.route.queryParamMap,
-    );
-
-    // subscribe to the single observable, giving us all the data we need
-    return combined$
-      .pipe(
-        tap(([user, isLocalAuthority, isProvincialAdmin]) => {
-          this.currentUser = user;
-          this.isLocalAuthority = isLocalAuthority;
-          this.isProvincialAdmin = isProvincialAdmin;
-        }),
-        switchMap(([user, isLocalAuthority, isProvincialAdmin, queryParams]) => {
-          // get the organization matching the query parameter passed in (if any)
-          if (isProvincialAdmin) {
-            const orgId = this.uniqueKeyService.getKey();
-            return this.getAnyOrganization(orgId);
-          }
-          // for local authorities, we know what org they belong to
-          if (isLocalAuthority) {
-            const volunteerId = user.contactid;
-            return this.getVolunteerOrganization(volunteerId);
-          }
-        }),
-        tap(org => this.currentOrganization = org)
-      );
-  }
-
-  getAnyOrganization(orgId: string): Observable<Organization> {
-    // get an organization by its ID
-    if (!orgId) {
-      // if there is no organization ID included return null
-      return of(null as Organization);
-    }
-    // if there is an organization id included look up the organization in the service by its ID
-    return this.organizations.getOrganizationById(orgId);
-  }
-
-  getVolunteerOrganization(volunteerId: string): Observable<Organization> {
-    // get the organization associated with the volunteer
-    if (!volunteerId) {
-      // if there is no volunteer ID included return null because a null volunteer cannot have an organization
-      return of(null as Organization);
+  copyProperties(obj: {}): {} {
```

Comment:
For future reference, there's probably a lodash function that does this for you.